esp-hal-embassy/src/lib.rs

```diff
@@ -121,6 +121,23 @@ impl<const N: usize> TimerCollection for &'static mut [Timer; N] {
     }
 }
 
+macro_rules! impl_array {
+    ($n:literal) => {
+        impl<T> TimerCollection for [T; $n]
+        where
+            T: IntoErasedTimer,
+        {
+            fn timers(self) -> &'static mut [Timer] {
+                mk_static!([Timer; $n], self.map(|t| Timer::new(t.into())))
```

Comment:
I'm guessing you initially tried const generics instead of a macro. Out of interest, where did the compiler complain?

Reply:
https://rust.godbolt.org/z/rE3c6aKvY this should be illustrative :)